visma/functions/variable.py

```diff
@@ -77,3 +77,177 @@ def integrate(self, wrtVar):
 
     def calculate(self, val):
         return self.coefficient * ((val**(self.power)))
+
+    def __radd__(self, other):
+        return self + other
+
+    def __add__(self, other):
+        from visma.functions.constant import Constant
+        if isinstance(other, Variable):
+            sortedValuesSelf = self.value.sort()
+            sortedValuesOther = other.value.sort()
+            if (self.power == other.power) & (sortedValuesSelf == sortedValuesOther):
+                if self.before == '-':
+                    self.coefficient -= other.coefficient
+                else:
+                    self.coefficient += other.coefficient
+                result = Variable()
+                if self.coefficient != 0:
+                    result.scope = self.scope
+                    result.power = self.power
+                    result.coefficient = self.coefficient
+                return self
+        elif isinstance(other, Constant):
+            expression = Expression()
+            expression.tokens = [self]
+            expression.tokens.extend([Plus(), other])
+            self.type = 'Expression'
+            self = expression
+            return expression
+        elif isinstance(other, Expression):
+            expression = Expression()
+            expression.tokens = [self]
+            for i, token in enumerate(other.tokens):
+                if isinstance(token, Variable):
+                    if token.power == self.power:
+                        self.coefficient += other.tokens[i].coefficient
+                    else:
+                        expression.tokens.extend([Plus(), Variable(token)])
+                elif isinstance(token, Constant):
+                    expression.tokens.extend([Plus(), Constant(token.calculate()*1)])
```

Comment:
Include both plus and minus cases.
```suggestion
                    expression.tokens.extend([Plus(), token)
```